### PR TITLE
Amp analytics: Add SUM global function

### PR DESCRIPTION
**Pull request by pomeroyr**
For use in batched pings, adds passed values together. Example:

SUM(namespace1, 5) = 5
SUM(namespace1, 10) = 15
SUM(namespace2, 2) = 2

Documentation in PR #12249

### CODE REVIEW

**Comment by pomeroyr** [comment]
@zhouyx @dvoytenko @lannka

**Comment by pomeroyr** [comment]
Friendly ping. @zhouyx @dvoytenko @lannka

**Comment by lannka** [comment]
Not sure if those questions have been raised, but:

- Do you need to sum up float number?
- If not, can we reload the COUNTER function by providing an additional param?

**Comment by lannka** [comment]
COUNTER doesn't need to +1 all the time. It also makes a lot of sense for a COUNTER to count faster.

I'm not sure about your exact use case. If the user has to provide the explicit integer in the URL, I feel it is more like a "COUNTER". For a SUM function, I'd expect it to take vars in the expression.

**Comment by pomeroyr** [comment]
@lannka Our use case would generally look something like this:
'request': 'https://e.com?d=SUM(${requestId},${timerDuration})'

**Comment by lannka** [comment]
@pomeroyr that sounds reasonable.

File: extensions/amp-a4a/0.1/a4a-variable-source.js [hunk 1]
```diff
@@ -27,6 +27,7 @@ import {user} from '../../../src/log';
 const WHITELISTED_VARIABLES = [
   'RANDOM',
   'COUNTER',
+  'SUM',
```
Comment (lannka) [inline extensions/amp-a4a/0.1/a4a-variable-source.js]: can you sort your new function in alphabetic order?
same below

Comment (pomeroyr) [reply extensions/amp-a4a/0.1/a4a-variable-source.js]: Done. I wasn't sure how to sort one item in an unsorted list, so I just sorted everything.

**Comment by lannka** [comment]
decided to go with a different approach to fulfill the original requirements from RichMedia: #12432

**Pull request closed without merging**